drs_downloader/cli.py

```diff
@@ -1,11 +1,183 @@
+@click.option("--manifest_path", "-m", show_default=True,
+              help="Path to manifest tsv.")
+@click.option('--drs_header', default='ga4gh_drs_uri', show_default=True,
+              help='The column header in the TSV file associated with the DRS URIs.'
+                   'Example: pfb:ga4gh_drs_uri')
+def mock(silent: bool, destination_dir: str, manifest_path, drs_header):
+    """Generate test files locally, without the need for server."""
+
+    #
+    # get ids from manifest
+    ids_from_manifest = _extract_tsv_info(Path(manifest_path), drs_header)
+
+    # perform downloads with a mock drs client
+    _perform_downloads(destination_dir, MockDrsClient(), ids_from_manifest, silent)
+
+
+@cli.command()
+@click.option("--silent", "-s", is_flag=True, show_default=True, default=False, help="Display nothing.")
+@click.option("--destination_dir", "-d", show_default=True,
+              default="/tmp/testing", help="Destination directory.")
+@click.option("--manifest_path", "-m", show_default=True,
+              help="Path to manifest tsv.")
+@click.option('--drs_header', default=None, help='The column header in the TSV file associated with the DRS URIs.'
+              'Example: pfb:ga4gh_drs_uri')
+def terra(silent: bool, destination_dir: str, manifest_path: str, drs_header: str):
+    """Copy files from terra.bio"""
+
+    # get ids from manifest
+    ids_from_manifest = _extract_tsv_info(Path(manifest_path), drs_header)
+
+    # perform downloads with a terra drs client
+    _perform_downloads(destination_dir, TerraDrsClient(), ids_from_manifest, silent)
+
+
+@cli.command()
+@click.option("--silent", "-s", is_flag=True, show_default=True, default=False, help="Display nothing.")
+@click.option("--destination_dir", "-d", show_default=True,
+              default="/tmp/testing", help="Destination directory.")
+@click.option("--manifest_path", "-m", show_default=True,
+              help="Path to manifest tsv.")
+@click.option('--drs_header', default='ga4gh_drs_uri', show_default=True,
+              help='The column header in the TSV file associated with the DRS URIs.'
+                   'Example: pfb:ga4gh_drs_uri')
+@click.option('--api_key_path', show_default=True,
+              help='Gen3 credentials file')
+@click.option('--endpoint', show_default=True, required=True,
+              help='Gen3 endpoint')
+def gen3(silent: bool, destination_dir: str, manifest_path: str, drs_header: str, api_key_path: str, endpoint: str):
+    """Copy files from gen3 server."""
+    # read from manifest
+    ids_from_manifest = _extract_tsv_info(Path(manifest_path), drs_header)
+
+    _perform_downloads(destination_dir,
+                       Gen3DrsClient(api_key_path=api_key_path, endpoint=endpoint),
+                       ids_from_manifest,
+                       silent)
+
+
+def _perform_downloads(destination_dir, drs_client, ids_from_manifest,  silent):
+    """Common helper method to run downloads."""
+    # verify parameters
```

Comment:
@mbaumann-broad  : just double checking on this one.  Do you mean logging an INFO message with the full path name of the destination file?

Reply:
@bwalsh 
If the current user output looks like this:
```
$ drs_downloader terra -m tests/fixtures/terra-data.tsv -d DATA
100%|████████████████████████████████| 10/10 [00:00<00:00, 56148.65it/s]

2022-11-21 16:56:49,595 ('HG03873.final.cram.crai', 'OK', 1351946, 1)
2022-11-21 16:56:49,595 ('HG04209.final.cram.crai', 'OK', 1338980, 1)
2022-11-21 16:56:49,595 ('HG02142.final.cram.crai', 'OK', 1405543, 1)
...
```
It would be helpful to output the full path to the destination directory early on.
Something like this:
```
$ drs_downloader terra -m tests/fixtures/terra-data.tsv -d DATA
Downloading to: /home/mbaumann/anvil/DATA
100%|████████████████████████████████| 10/10 [00:00<00:00, 56148.65it/s]

2022-11-21 16:56:49,595 ('HG03873.final.cram.crai', 'OK', 1351946, 1)
2022-11-21 16:56:49,595 ('HG04209.final.cram.crai', 'OK', 1338980, 1)
2022-11-21 16:56:49,595 ('HG02142.final.cram.crai', 'OK', 1405543, 1)
...
```
This helps confirm to users that the data is being downloaded to their intended destination, and (hopefully) help ensure enough space will be available on the storage volume for the download to complete successfully, etc.

Reply:
[Fixed](https://github.com/anvilproject/drs_downloader/blob/feature/manager/drs_downloader/cli.py#L97-L98) and [tested](https://github.com/anvilproject/drs_downloader/blob/feature/manager/tests/integration/test_cli_options.py#L21)

Reply:
@lbeckman314, two corrections needed:
- The intention was for the "Downloading to: ..." message to be displayed where the user can see it, not written to a log file.
- The full/absolute path should be displayed, not the just a relative path as the user may have provided as the value of the `-d` option

Please correct these.